File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        if len(nums) == 1: return nums[0]
        nums.sort()

        i = 0
        majority = ceil(len(nums) / 2)
        print(majority)

        while i < len(nums):

            c_num = nums[i]
            c_freq = 1
            i += 1
            while i < len(nums) and nums[i] == c_num:
                i += 1
                c_freq += 1
                if c_freq >= majority:
                    return c_num
        
        return 0
             
        

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        
        int maj = ceil((float) nums.size() / 2);
        cout  << maj;
        unordered_map<int, int> freq;
        for (auto num: nums) {
            freq[num]++;
            if (freq[num] == maj) {
                return num;
            }
        }
        
        return -1;
        
    }
};